plugins/woocommerce/includes/class-woocommerce.php

```diff
@@ -585,6 +585,7 @@ public function includes() {
 		include_once WC_ABSPATH . 'includes/queue/class-wc-action-queue.php';
 		include_once WC_ABSPATH . 'includes/queue/class-wc-queue.php';
 		include_once WC_ABSPATH . 'includes/admin/marketplace-suggestions/class-wc-marketplace-updater.php';
+		include_once WC_ABSPATH . 'includes/admin/class-wc-admin-marketplace-promotions.php';
```

Comment:
This means `WC_Admin_Marketplace_Promotions::init` is run for every request, even front end Ajax and Cron requests. It seems a shame that we have to do this simply because some users apparently run cron without setting `DOING_CRON`. But I can see there's not much else we can do to account for that case. And we have enough conditions in `init` to mean this should not be too heavy.

Reply:
This is now superseded by the new method of fetching.